engine/dlib/src/dlib/http_client.cpp

```diff
@@ -1098,9 +1113,22 @@ if (sock_res != dmSocket::RESULT_OK)\
         }
     }
 
+    Result SetCacheKey(HClient client, const char* key)
+    {
+        uint32_t n = dmSnPrintf(client->m_CacheKey, sizeof(client->m_CacheKey), "%s", key);
+        if (n < sizeof(client->m_CacheKey))
+            return RESULT_OK;
+        return RESULT_INVAL;
+    }
```

Comment:
I opted for setting the cache key before the Get() call, as we may need it directly inside that function, and the Get() call doesn't have access to the headers.